src/Plutus/Types/Value.purs

```diff
@@ -0,0 +1,225 @@
+module Plutus.Types.Value
+  ( Value
+  , getValue
+  , singleton
+  , singleton'
+  , valueOf
+  , lovelaceValueOf
+  , scale
+  , symbols
+  , isZero
+  , negation
+  , split
+  , unionWith
+  , geq
+  , gt
+  , leq
+  , lt
+  , eq
+  , CurrencySymbol
+  , getCurrencySymbol
+  , mkCurrencySymbol
+  , adaSymbol
+  , flattenValue
+  ) where
+
+import Prelude hiding (eq)
+
+import Control.Apply (lift3)
+import Data.Array (concatMap)
+import Data.BigInt (BigInt)
+import Data.Foldable (all)
+import Data.Maybe (Maybe(Nothing), fromMaybe)
+import Data.These (These(Both, That, This), these)
+import Data.Tuple.Nested (type (/\), (/\))
+import FromData (class FromData, fromData)
+import Serialization.Hash (scriptHashFromBytes)
+import ToData (class ToData, toData)
+import Types.ByteArray (ByteArray)
+import Types.TokenName (TokenName, adaToken, mkTokenName)
+import Plutus.Types.AssocMap (Map(Map)) as Plutus
+import Plutus.Types.AssocMap (singleton, lookup, keys, union, mapThese) as Plutus.Map
+
+newtype Value = Value (Plutus.Map CurrencySymbol (Plutus.Map TokenName BigInt))
+
+instance Show Value where
+  show (Value mp) = "(PlutusValue " <> show mp <> ")"
+
+instance Semigroup Value where
+  append = unionWith add
+
+instance Monoid Value where
+  mempty = Value (Plutus.Map [])
+
+--------------------------------------------------------------------------------
```

Comment:
We could alternatively define an `Eq` instance using our own `eq` for convenience? I'll leave that down to you.